daemon/api.go

```diff
@@ -1539,16 +1539,25 @@ func getSnapConf(c *Command, r *http.Request, user *auth.UserState) Response {
 	snapName := vars["name"]
 
 	keys := splitQS(r.URL.Query().Get("keys"))
-	if len(keys) == 0 {
-		return BadRequest("cannot obtain configuration: no keys supplied")
-	}
 
 	s := c.d.overlord.State()
 	s.Lock()
 	tr := config.NewTransaction(s)
 	s.Unlock()
 
 	currentConfValues := make(map[string]interface{})
+	// Special case - return root document
+	if len(keys) == 0 {
```

Comment:
Instead of duplicating all of this, we can keep both cases under the same loop by doing:
```
if len(keys) == 0 {
    keys = []string{""}
}
```
and then below
```
if key == "" {
        if len(keys) > 1 {
                return BadRequest("keys contains zero-length string", err)
        }
        for k, v := range value {
                currentConfValues[key] = value
        }
} else {
        currentConfValues[key] = value
}
```